pilot/pkg/model/validation.go

```diff
@@ -1805,7 +1815,11 @@ func validateAuthNPolicyTarget(target *authn.TargetSelector) (errs error) {
 		return
 	}
 
-	errs = appendErrors(errs, validateHost(target.Name))
+	// AuthN policy target (host)name must be a shortname
+	if !IsDNS1123Label(target.Name) {
```

Comment:
Why do you prefer to use DNS label than the host name?

Reply:
We restrict the target name must be a short name only, so that fullname can be constructed correctly from policy's namespace and domain. We don't want a fancy wildcards neither. I think IsDNS1123Label is a more appropriate function to validate shortname (it's used to validate each part of hostname break down like this https://github.com/istio/istio/blob/master/pilot/pkg/model/validation.go#L153)